lib/hiera/backend.rb

```diff
@@ -15,7 +15,7 @@ def initialize(wrapped)
 
       def lookup(key, scope, order_override, resolution_type, context)
         Hiera.debug("Using Hiera 1.x backend API to access instance of class #{@wrapped.class.name}. Lookup recursion will not be detected")
-        value = @wrapped.lookup(key, scope, order_override, resolution_type)
+        value = @wrapped.lookup(key, scope, order_override, resolution_type.is_a?(Hash) ? :hash : resolution_type)
```

Comment:
wonder if this is correct - if the resolution_type is a hash isn't mapping the strategy the right thing to do? (confused...)